tools/describezones/main.go

```diff
@@ -0,0 +1,178 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"go/format"
+	"log"
+	"os"
+	"path"
+	"text/template"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/zclconf/go-cty/cty"
+	"google.golang.org/api/compute/v1"
+)
+
+func main() {
+	flag.Parse()
+	provider := flag.Arg(0)
+	switch provider {
+	case "gcp":
+		describeGCPZones()
+	default:
+		describeAWSZones()
+	}
+}
+
+func describeAWSZones() {
+	cfg, err := config.LoadDefaultConfig(context.Background())
+	if err != nil {
+		log.Fatalf("error loading aws config %s", err)
+	}
+
+	svc := ec2.NewFromConfig(cfg)
+
+	resp, err := svc.DescribeRegions(context.Background(), &ec2.DescribeRegionsInput{
+		AllRegions: aws.Bool(true),
+	})
+	if err != nil {
+		log.Fatalf("error describing ec2 regions %s", err)
+	}
+	m := make(map[string]map[string]cty.Value)
+	for _, region := range resp.Regions {
+		name := *region.RegionName
+		regionalConf, err := config.LoadDefaultConfig(context.Background(),
+			config.WithRegion(name),
+		)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		regionalSvc := ec2.NewFromConfig(regionalConf)
+
+		result, err := regionalSvc.DescribeAvailabilityZones(context.Background(), &ec2.DescribeAvailabilityZonesInput{
+			AllAvailabilityZones: aws.Bool(true),
```

Comment:
Looks like this returns the wavelength and other zones as well. I guess we only want these if the user has `all_available_zones` on the `data.aws_availability_zones` resource. I guess this is difficult because we have no way of knowing if the user has opted into them or not. So maybe if we find this is an issue with anyone in the future, we can always assume they've not opted into these zones as a default.